File: app/src/main/java/com/buzzybrains/mvvmarchitecture/asynctask/DeleteAllNoteAsyncTask.java
package com.buzzybrains.mvvmarchitecture.asynctask;

import android.os.AsyncTask;

import com.buzzybrains.mvvmarchitecture.data.NoteDao;

public class DeleteAllNoteAsyncTask extends AsyncTask<Void, Void, Void> {

    private NoteDao noteDao;

    public DeleteAllNoteAsyncTask(NoteDao noteDao) {
        this.noteDao = noteDao;

    }

    @Override
    protected Void doInBackground(Void... voids) {

        noteDao.deleteAllNotes();

        return null;

    }
}

File: app/src/main/java/com/buzzybrains/mvvmarchitecture/asynctask/DeleteAllNoteAsyncTask.kt
package com.buzzybrains.mvvmarchitecture.asynctask

import android.os.AsyncTask

import com.buzzybrains.mvvmarchitecture.data.NoteDao

class DeleteAllNoteAsyncTask(private val noteDao: NoteDao) : AsyncTask<Void, Void, Void>() {

    override fun doInBackground(vararg voids: Void): Void? {

        noteDao.deleteAllNotes()

        return null

    }
}

